mypy/server/astdiff.py

```diff
@@ -83,7 +83,10 @@ def is_similar_node_shallow(n: SymbolTableNode, m: SymbolTableNode) -> bool:
                 nn.is_newtype == mn.is_newtype and
                 is_same_mro(nn.mro, mn.mro))
     if isinstance(n.node, Var) and isinstance(m.node, Var):
-        return is_identical_type(n.node.type, m.node.type)
+        if n.node.type is None and m.node.type is None:
```

Comment:
Wait, I misread the code. If one of them is `None` and the other isn't, this will return `False` (without calling `is_identical_type()`). But if you're now no longer sure that it should return `True` if both are `None`, then @JukkaL must clarify.